src/renderer/ducks/validator/sagas.ts

```diff
@@ -234,18 +234,23 @@ function* validatorInfoUpdater(
 ): Generator<
     SelectEffect | PutEffect | CancelEffect | RaceEffect<TakeEffect> | Promise<undefined | bigint> | Promise<void>,
     void,
-    IValidator & [ReturnType<typeof removeActiveValidator>, ReturnType<typeof finalizedEpoch>] & (undefined | bigint)
+    IValidator & [ReturnType<typeof removeActiveValidator>, ReturnType<typeof newEpoch>] & (undefined | bigint)
 > {
     while (true) {
         try {
-            const [cancelAction, {payload}] = yield race([take(removeActiveValidator), take(finalizedEpoch)]);
+            const [cancelAction, {payload}] = yield race([take(removeActiveValidator), take(newEpoch)]);
             if (cancelAction && cancelAction.payload === publicKey) {
                 yield cancel();
             }
 
             const validator = yield select(getValidator, {publicKey});
             if (validator.beaconNodes.includes(payload.beacon)) {
-                const balance = yield getValidatorBalance(publicKey, network, payload.beacon);
+                const balance = yield getValidatorBalance(
+                    publicKey,
+                    network,
+                    payload.beacon,
+                    BigInt(payload.epoch * 32),
```

Comment:
ugh, heck:
- I totally forgot validator is fetched by slot (don't convert to epoch and back to slot here, just send slot here)
- slot isn't BigInt